test/tools/utils.js

```diff
@@ -0,0 +1,236 @@
+ * @param {Number} min
+ * @param {Number} max
+ * @param {Number} size
+ * @param {String} [type='float64']
+ * @param {String} [sign='mixed']
+ * @return {Array<Number>}
+ */
+function getRandomNumbers(min, max, size, type = 'float64', sign='mixed') {
+  const data = new Array(size);
+  for (let i = 0; i < size; i++) {
+    data[i] = getRandom(min, max, type, sign);
+  }
+  return getPrecisionData(data, type);
+}
+
+/**
+ * Prepare input data by specified config of inputsDataInfo, dataFile, min,
+ * max parameters .
+ * @param {Object} inputsDataInfo information object for input data
+ * @param {String} dataFile saved data file path
+ * @param {Number} min
+ * @param {Number} max
+ * @return {Object}
+ */
+function prepareInputsData(inputsDataInfo, dataFile, min, max) {
+  const dstDataDict = {inputsData: {}};
+  let srcDataDict = {};
+  if (fs.existsSync(dataFile)) {
+    srcDataDict = utils.readJsonFile(dataFile);
+  }
+  for (const source in inputsDataInfo) {
+    // reserve last input data when generating new required input data
+    if (srcDataDict['inputsData'] !== undefined &&
+        srcDataDict['inputsData'][source] !== undefined) {
+      dstDataDict['inputsData'][source] = srcDataDict['inputsData'][source];
+    } else {
+      const targetDataInfo = inputsDataInfo[source];
+      if (targetDataInfo.data !== undefined) {
+        const srcDataInfo = inputsDataInfo[targetDataInfo.data];
+        const inputTensor = new Tensor(
+            srcDataInfo.shape, dstDataDict['inputsData'][targetDataInfo.data]);
+        let outputTensor;
+        if (targetDataInfo.processCategory === 'transpose') {
+          outputTensor = transpose(
+              inputTensor, {permutation: targetDataInfo.permutation});
+        } else if (targetDataInfo.processCategory === 'negative') {
+          outputTensor = neg(inputTensor);
+        }
+        console.log(`source ${source}`);
+        dstDataDict['inputsData'][source] = outputTensor.data;
+      } else {
+        const total = sizeOfShape(targetDataInfo.shape);
+        const sign = targetDataInfo.sign;
+        const type = targetDataInfo.type;
+        if (targetDataInfo.dataRange) {
+          min = targetDataInfo.dataRange[0];
+          max = targetDataInfo.dataRange[1];
+        }
+        const generatedNumbers =
+            utils.getRandomNumbers(min, max, total, type, sign);
+        dstDataDict['inputsData'][source] = generatedNumbers;
+      }
+    }
+  }
+  return dstDataDict;
+}
+
+/**
+ * Get JSON object from specified JSON file.
+ * @param {String} filePath
+ * @return {Object}
+ */
+function readJsonFile(filePath) {
+  let inputFile;
+  if (path.isAbsolute(filePath)) {
+    inputFile = filePath;
+  } else {
+    inputFile =
+        path.join(path.dirname(process.argv[1]), filePath);
+  }
+  const content = fs.readFileSync(inputFile).toString();
+  const jsonDict = JSON.parse(
+      content.replace(/\\"|"(?:\\"|[^"])*"|(\/\/.*|\/\*[\s\S]*?\*\/)/g,
```

Comment:
What is this regex doing? Is it removing comments?

```suggestion
      content.replace(/\\"|"(?:\\"|[^"])*"|(\/\/.*|\/\*[\s\S]*?\*\/)/g, // remove comments
```